src/stores/store/keys/local/local.go

```diff
@@ -218,16 +214,26 @@ func (s *Store) Sign(ctx context.Context, id string, data []byte, algo *entities
 		return nil, errors.DependencyFailureError(errMessage)
 	}
 
+	var signature []byte
 	switch {
 	case algo.Type == entities2.Eddsa && algo.EllipticCurve == entities2.Babyjubjub:
-		return s.signEDDSA(privkey, data)
+		signature, err = eddsa.SignBabyjubjub(privkey, data)
 	case algo.Type == entities2.Ecdsa && algo.EllipticCurve == entities2.Secp256k1:
-		return s.signECDSA(privkey, data)
+		signature, err = ecdsa.SignSecp256k1(privkey, data)
+	case algo.Type == entities2.Eddsa && algo.EllipticCurve == entities2.X25519:
+		signature, err = eddsa.SignX25519(privkey, data)
 	default:
 		errMessage := "signing algorithm and curve combination not supported for signing"
 		logger.With("algorithm", algo.Type, "curve", algo.EllipticCurve).Error(errMessage)
 		return nil, errors.InvalidParameterError(errMessage)
 	}
+
+	if err != nil {
```

Comment:
I think doing `return` directly in the switches was correct because these functions should not come from a package and return a specific error type. If you want to move them to a package, they should return a standard error and this `if err != nil` should return the specific error (but you won't know what happened)

Reply:
Discussed